File: android/src/main/kotlin/com/clashsiing/flutter_sing_box/utils/SingBoxManager.kt
package com.clashsiing.flutter_sing_box.utils

import com.tencent.mmkv.MMKV

object SingBoxManager {
    private val mmkv = MMKV.mmkvWithID("cs-sing-box", MMKV.MULTI_PROCESS_MODE)
    fun getProfile(): String? {
        return mmkv.decodeString("profile")
    }
}

File: lib/src/settings/sing_box_manager.dart
import 'package:mmkv/mmkv.dart';

class SingBoxManager {
  static final SingBoxManager _instance = SingBoxManager._internal();
  factory SingBoxManager() => _instance;

  late MMKV _mmkv;
  SingBoxManager._internal() {
    _mmkv = MMKV("cs-sing-box", mode: MMKVMode.MULTI_PROCESS_MODE);
  }

  String? get profile {
    return _mmkv.decodeString("profile");
  }

  set profile(String profile) {
    _mmkv.encodeString("profile", profile);
  }

}
final singBoxManager = SingBoxManager();